### PR TITLE
REGR: SeriesGroupBy where index has a tuple name fails

### LINKED ISSUE #37755: BUG: Series.groupby() fails in pandas 1.1.4 when index has tuple name.
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd

print("Test 1")
a = pd.Series([1,2,3,4], index=[1,1,2,2], name=("a", "a"))
a.index.name = ("b", "b")
print(a)
print(a.index)
print(a.groupby(level=0).last())

print("Test 2")
a = pd.Series([1,2,3,4], index=[2,3,4,5], name=("a", "a"))
b = pd.Series([1,1,2,2], index=[2,3,4,5], name=("b", "b"))
a.index = b.reindex(a.index)
print(a)
print(a.index)
print(a.groupby(level=0).last())
```

#### Problem description

In pandas 1.1.2 this works fine. While it crashes in pandas 1.1.4. The problem seems related to the tuple index names. The output is:
``` 
Test 1
(b, b)
1    1
1    2
2    3
2    4
Name: (a, a), dtype: int64
Int64Index([1, 1, 2, 2], dtype='int64', name=('b', 'b'))
Traceback (most recent call last):
  File "testcase.py", line 8, in <module>
    print(a.groupby(level=0).last())
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/series.py", line 1735, in groupby
    return SeriesGroupBy(
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/groupby/groupby.py", line 525, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/groupby/grouper.py", line 773, in get_grouper
    if is_in_obj(gpr):  # df.groupby(df['name'])
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/groupby/grouper.py", line 765, in is_in_obj
    return gpr is obj[gpr.name]
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/series.py", line 888, in __getitem__
    result = self._get_value(key)
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/series.py", line 989, in _get_value
    loc = self.index.get_loc(label)
  File "/home/burk/.local/share/virtualenvs/timeseries-Gj3kjmJv/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2895, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 96, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 118, in pandas._libs.index.IndexEngine._get_loc_duplicates
TypeError: only integer scalar arrays can be converted to a scalar index
```

#### Expected Output
This is the output in 1.1.2:
```
Test 1
(b, b)
1    1
1    2
2    3
2    4
Name: (a, a), dtype: int64
Int64Index([1, 1, 2, 2], dtype='int64', name=('b', 'b'))
(b, b)
1    2
2    4
Name: (a, a), dtype: int64

Test 2
(b, b)
1    1
1    2
2    3
2    4
Name: (a, a), dtype: int64
Int64Index([1, 1, 2, 2], dtype='int64', name=('b', 'b'))
(b, b)
1    2
2    4
Name: (a, a), dtype: int64
```

#### Output of ``pd.show_versions()``

<details>
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.8.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.8.0-26-generic
Version          : #27-Ubuntu SMP Wed Oct 21 22:29:16 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.4
numpy            : 1.19.4
pytz             : 2020.4
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 44.0.0
Cython           : 0.29.17
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.6.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : None
IPython          : 7.19.0
pandas_datareader: 0.9.0
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : None
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : 2.0.0
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.20
tables           : None
tabulate         : 0.8.7
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.51.2
</details>

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #37755
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

The groupby code that calls `Series.__getitem__` catches `KeyError` and `TypeError`. If a tuple is fed into an `Index` that is not a `MultiIndex`, we were raising a `ValueError` but `KeyError` seems more appropriate here and also fixes the regression.

While working on this, I found that we raise a `KeyError` inappropriately if an index of tuples contains duplicates, however this exists back in 1.0.x and so is not part of the regression. I've opened #37800 for this.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2146,3 +2146,13 @@ def test_groupby_duplicate_columns():
     result = df.groupby([0, 0, 0, 0]).min()
     expected = DataFrame([["e", "a", 1]], columns=["A", "B", "B"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_series_with_tuple_name():
+    # GH 37755
+    s = Series([1, 2, 3, 4], index=[1, 1, 2, 2], name=("a", "a"))
+    s.index.name = ("b", "b")
+    result = s.groupby(level=0).last()
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_groupby.py:2155]: is anything about this specific to `last`?

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2146,3 +2146,13 @@ def test_groupby_duplicate_columns():
     result = df.groupby([0, 0, 0, 0]).min()
     expected = DataFrame([["e", "a", 1]], columns=["A", "B", "B"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_series_with_tuple_name():
+    # GH 37755
+    s = Series([1, 2, 3, 4], index=[1, 1, 2, 2], name=("a", "a"))
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_groupby.py:2153]: nitpick: can you call this `ser` or something else that isnt 1 letter

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/groupby/test_groupby.py:2155]: No, in fact the regression would raise in the call to `groupby`, I just took last from the issue that was raised. Should something else be done here?

Comment (jbrockmendel) [reply pandas/tests/groupby/test_groupby.py:2155]: > Should something else be done here?

no, just curious

**Comment by jreback** [comment]
thanks @rhshadrach keem em coming!

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**